File: 2015/Problem6.java
public class Problem6{
	public static void main(String args[]){
		int sumSquares = 0, sum = 0;

		for (int i = 1; i < 101; i++){
			sumSquares += (i * i);
			sum += i;
		}

		int squareSum = (sum * sum);

		System.out.println("The difference is " + (squareSum - sumSquares));
	}
}

File: problem6.js
function range(limit) {
  let array = []
  for (i = 0; i < limit; i++) {
    array[i] = i + 1
  }
  return array
}

function sumSquares(array) {
  let sum = 0
  for (let i = 0; i < array.length; i++) {
    sum += Math.pow(array[i], 2)
  }
  return sum
}

function squareSum(array) {
  let sum = 0
  for (let i = 0; i < array.length; i++) {
    sum += array[i]
  }
  return Math.pow(sum, 2)
}

function problem6() {
  let array = range(100)
  
  const sumSQ = sumSquares(array)
  const sqSum = squareSum(array)
  return sumSQ - sqSum
}

const startTime = Date.now()
const number = problem6()
const endTime = Date.now()
console.log('number: ', number)
const runningTime = endTime - startTime
console.log(`Running time (s): ${runningTime / 1000}`)
